src/tests/test_magic.py

```diff
@@ -1240,29 +1274,27 @@ def test_save_with_number_table(
 
 
 def test_save_with_non_existing_with(ip):
-    out = ip.run_cell(
-        "%sql --with non_existing_sub_query " "SELECT * FROM non_existing_sub_query"
+    with pytest.raises(UsageError) as excinfo:
+        ip.run_cell(
+            "%sql --with non_existing_sub_query SELECT * FROM non_existing_sub_query"
+        )
+
+    assert '"non_existing_sub_query" is not a valid snippet identifier.' in str(
+        excinfo.value
     )
-    assert isinstance(out.error_in_exec, UsageError)
+    assert excinfo.value.error_type == "UsageError"
```

Comment:
why do we check the error type here? we know it's `UsageError` from line 1277

Reply:
added explanation to the top of `exceptions.py`